e2e/tests/test_smoke.py

```diff
@@ -0,0 +1,146 @@
+
+        # TODO check if CRD is registered instead
+        Utils.wait_for_pod_start(k8s_api, 'name=postgres-operator', cls.RETRY_TIMEOUT_SEC)
+
+        # HACK around the lack of Python client for the acid.zalan.do resource
+        subprocess.run(["kubectl", "create", "-f", "manifests/minimal-postgres-manifest.yaml"])
+
+        Utils.wait_for_pod_start(k8s_api, 'spilo-role=master', cls.RETRY_TIMEOUT_SEC)
+
+    @timeout_decorator.timeout(TEST_TIMEOUT_SEC)
+    def test_master_is_unique(self):
+        """
+           Check that there is a single pod in the k8s cluster with the label "spilo-role=master".
+        """
+        k8s = K8sApi()
+        labels = 'spilo-role=master,version=acid-minimal-cluster'
+
+        num_of_master_pods = Utils.count_pods_with_label(k8s, labels)
+        self.assertEqual(num_of_master_pods, 1, "Expected 1 master pod, found {}".format(num_of_master_pods))
+
+    @timeout_decorator.timeout(TEST_TIMEOUT_SEC)
+    def test_scaling(self):
+        """
+           Scale up from 2 to 3 pods and back to 2 by updating the Postgres manifest at runtime.
+        """
+        k8s = K8sApi()
+        labels = "version=acid-minimal-cluster"
+
+        Utils.wait_for_pg_to_scale(k8s, 3, self.RETRY_TIMEOUT_SEC)
+        self.assertEqual(3, Utils.count_pods_with_label(k8s, labels))
+
+        # TODO `kind` pods may stuck in the Terminating state for a few minutes; reproduce/file a bug report
+        Utils.wait_for_pg_to_scale(k8s, 2, self.RETRY_TIMEOUT_SEC)
+        self.assertEqual(2, Utils.count_pods_with_label(k8s, labels))
+
+
+class K8sApi:
+
+    def __init__(self):
+
+        # https://github.com/kubernetes-client/python/issues/309
+        warnings.simplefilter("ignore", ResourceWarning)
+
+        self.config = config.load_kube_config()
+        self.k8s_client = client.ApiClient()
+        self.core_v1 = client.CoreV1Api()
+        self.crd = client.CustomObjectsApi()
+        self.apps_v1 = client.AppsV1Api()
+
+
+class Utils:
+
+    @staticmethod
+    def wait_for_pod_start(k8s_api, pod_labels, retry_timeout_sec):
+        pod_phase = 'No pod running'
+        with Halo(text="Wait for the pod '{}' to start. Pod phase: {}".format(pod_labels, pod_phase), spinner='dots'):
+            while pod_phase != 'Running':
+                pods = k8s_api.core_v1.list_namespaced_pod('default', label_selector=pod_labels).items
+                if pods:
+                    pod_phase = pods[0].status.phase
+            time.sleep(retry_timeout_sec)
+
+    @staticmethod
+    def wait_for_pg_to_scale(k8s_api, number_of_instances, retry_timeout_sec):
+
+        body = {
+            "spec": {
+                "numberOfInstances": number_of_instances
+            }
+        }
+        _ = k8s_api.crd.patch_namespaced_custom_object("acid.zalan.do",
+                                                       "v1", "default", "postgresqls", "acid-minimal-cluster", body)
+
+        labels = 'version=acid-minimal-cluster'
```

Comment:
in other functions you've put the labels in the spec. I think it can also be done here. `'version=acid-minimal-cluster'` is also hard-coded in `test_scaling` from where `wait_for_pg_to_scale` is called